allennlp/modules/conditional_random_field.py

```diff
@@ -0,0 +1,195 @@
+"""
+Conditional random field
+"""
+from typing import List
+
+import torch
+from torch.autograd import Variable
+
+import allennlp.nn.util as util
+
+class ConditionalRandomField(torch.nn.Module):
+    """
+    This module uses the "forward-backward" algorithm to compute
+    the log-likelihood of its inputs assuming a conditional random field model.
+
+    See, e.g. http://www.cs.columbia.edu/~mcollins/fb.pdf
+
+    Parameters
+    ----------
+    num_tags : int, required
+        The number of tags.
+    """
+    def __init__(self,
+                 num_tags: int) -> None:
+        super().__init__()
+        self.num_tags = num_tags
+
+        # transitions[i, j] is the logit for transitioning from state i to state j.
+        self.transitions = torch.nn.Parameter(torch.randn(num_tags, num_tags))
```

Comment:
I still wonder if there's a better way to initialize these parameters.  I'd guess that `xavier_normal` is almost certainly better than `torch.randn`.